### PR TITLE
Bug/frontend auth

### PR DESCRIPTION
**Pull request by KristinaGru**
## 📝 Description

- added redirection logic for logged in/out user
- removed sidebar from authentication pages
- add alerts for errors during authentication
- fix login page submit button being enabled on mounting

## 🎉 Demos

image

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -46,24 +46,24 @@ const router = createBrowserRouter([
...
+    element: <LoginPage />,
+  },
+  {
+    path: '/forgot-password',
+    element: <ForgotPassword />,
+  },
+  {
+    path: '/reset-password',
+    element: <ResetPassword />,
+  },
```
Comment (kralle333) [inline frontend/src/App.tsx:68]: I think after this PR you should make a PR where the formatter + linter is run on the whole codebase. Reviewing PRs where actual changes are mixed in with formatting changes makes it harder to review.

File: frontend/src/pages/signUp/SignUp.tsx [hunk 2]
```diff
@@ -20,20 +21,36 @@ const SignUp = () => {
...
-    } catch (err) {
-      console.error(err);
+    } catch (e: any) {
+      if (e.message === 'Firebase: Error (auth/email-already-in-use).') {
+        setErrorMessage(`Email address '${data.email}' is already in use `);
+      } else if (e.message) {
+        setErrorMessage(e.message);
+      } else {
+        setErrorMessage('Something went wrong. Please try again');
+      }
```
Comment (kralle333) [inline frontend/src/pages/signUp/SignUp.tsx:41]: This code is duplicated in LoginPage.tsx, can you extract the error handling elsewhere?

File: frontend/src/pages/signUp/SignUp.tsx [hunk 3]
```diff
@@ -20,20 +21,36 @@ const SignUp = () => {
...
   const onSubmit: SubmitHandler<FormData> = async (data) => {
     try {
       await createUserWithEmailAndPassword(auth, data.email, data.password);
+      setErrorMessage('');
       navigate('/login');
-    } catch (err) {
-      console.error(err);
+    } catch (e: any) {
+      if (e.message === 'Firebase: Error (auth/email-already-in-use).') {
+        setErrorMessage(`Email address '${data.email}' is already in use `);
```
Comment (kralle333) [inline frontend/src/pages/signUp/SignUp.tsx]: not a good idea to do comparison on the message like that, if firebase decides to change the returned error your code breaks. There must be a better way to detect the error type. Is this how firebase want you to react on error types?

**Review by kralle333 — CHANGES_REQUESTED** [review]
Good job :) I only have some smaller comments

**Review by kralle333 — CHANGES_REQUESTED** [review]
Good job :) I only have some smaller comments

**Review by kralle333 — CHANGES_REQUESTED** [review]
Good job :) I only have some smaller comments

**Comment by KristinaGru** [comment]
@kralle333 thanks for review! 🙏 Switched to error code instead of error message check. As for changes in App.tsx, it's due to moving all authentication paths out of '/' paths children array, to fix the problem we've had of sidebar and header showing up in auth pages. Dev should be up to date now with all formatting changes 🙌

File: frontend/src/pages/signUp/SignUp.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ const SignUp = () => {
       setErrorMessage('');
       navigate('/login');
     } catch (e: any) {
-      if (e.message === 'Firebase: Error (auth/email-already-in-use).') {
+      if (e.code === 'auth/email-already-in-use') {
```
Comment (kralle333) [inline frontend/src/pages/signUp/SignUp.tsx]: Better :) But lets see if we can prevent using raw strings. It should be possible to use the constants here: https://firebase.google.com/docs/reference/js/auth#autherrorcodes

try importing like this:
`import { AuthErrorCodes } from 'firebase/auth'`

And then you should be able to use the constants for comparison:
`AuthErrorCodes.EMAIL_EXISTS`

**Review by kralle333 — CHANGES_REQUESTED** [review]

**Comment by KristinaGru** [comment]
Thanks for pointing that out! Switched it in the sign-up error check, but for login did not see anything that'd match the error code I want to check against. Found this issue, so will keep an eye on it if anything changes 🤓  https://github.com/firebase/firebase-js-sdk/issues/7661

**Comment by KristinaGru** [comment]
@kralle333 added a couple lines to solve the problem of unauthorized path showing for a second before being redirected to this PR. Hope it's ok 🤞

File: frontend/src/pages/login/LoginPage.tsx [hunk 5]
```diff
@@ -53,8 +56,15 @@ const LoginPage: React.FC = () => {
       const auth = getAuth();
       await setPersistence(auth, rememberMe ? browserLocalPersistence : browserSessionPersistence);
       await signInWithEmailAndPassword(auth, data.email, data.password);
-    } catch (e) {
-      console.error(e);
+      setErrorMessage('');
+    } catch (e: any) {
+      if (e.code === 'auth/invalid-login-credentials') {
```
Comment (kralle333) [inline frontend/src/pages/login/LoginPage.tsx:61]: You are missing using the constant here too :)

**Review by kralle333 — APPROVED** [review]
There is another constant you missed changing, but I won't let it block this PR. Good job!

File: frontend/src/pages/login/LoginPage.tsx (hunk 5 above)
Comment (kralle333) [reply frontend/src/pages/login/LoginPage.tsx:61]: ah i see you are aware and that its a issue with firebase. Very nice!

**Pull request merged**